### ISSUE TITLE
nginx: "listen ... http2" directive is deprecated

### ISSUE THREAD

**Issue by george-hopkins** [piku/piku#350]
nginx 1.26.1 does not work with piku because the config test fails at the moment:
```text
remote: Error: [nginx config] b'2024/06/12 12:09:24 [warn] 820#820: the "listen ... http2" directive is deprecated, use the "http2" directive instead in /home/piku/.piku/nginx/app1.conf:15\n2024/06/12 12:09:24 [warn] 820#820: the "listen ... http2" directive is deprecated, use the "http2" directive instead in /home/piku/.piku/nginx/app1.conf:16\n'
remote: Warning: removing broken nginx config.
```

It is recommended to use the following config.  However, the `http2` directive is only available from version 1.25.1.
```nginx
server {
    listen 443 ssl;
    http2 on;
    # ...
}
```

**Comment by rcarmo** [comment]
I'm all for getting rid of http2. Any new HTTP "modes" can be added in via the `NGINX_INCLUDE_FILE` setting. But Before I act on that, I need to know more about your setup - what is the distro and version you're using?

**Comment by george-hopkins** [comment]
I am using Alpine Linux (edge) and the master branch of piku. I fixed the issue temporarily by adding `grep -v 'directive is deprecated'` after `nginx -t`.

**Comment by chr15m** [comment]
Could add this fix to #348. :thinking:

**Comment by rcarmo** [comment]
I'm good with that. Just not sure if this affects Ubuntu/Debian LTS releases, since they're more conservative than Alpine edge (and that's a bit off the happy path for mainstream users).

**Comment by rcarmo** [comment]
I just merged #348 since I wanted to clear out outstanding pull requests, but yes, I think we need to have a look at this--I haven't had the time to check Ubuntu 24.04
